chameleon/src/main/java/org/hibernate/omm/ast/AbstractMongoSqlAstTranslator.java

```diff
@@ -3901,16 +3901,13 @@ protected void visitOrderBy(List<SortSpecification> sortSpecifications) {
         // If we have a query part for row numbering, there is no need to render the order by clause
```

Comment:
This class was mainly copied from Steve's class and it is not meant to be changed for each new Hibernate version bumping will end up overriding our customization here. Ultimately we should get rid of the aid of this class totally but for now it is not realistic and its existence acts as some reference.

So currently our Mongo customization mechanism goes as following:

- copied the `AbstractSqlAstTranslator` intact (with renaming to `AbstractMongoSqlAstTranslator`) here as the default implementation
- create one child class named `AbstractMongoQuerySqlTranslator` inheriting it to act as the class to override common MQL rendering stuff (like predicate rendering, common util methods, etc.)
- create two child classes inherting from the above `AbstractMongoQuerySqlTranslator`, focusing on pure query (`MongoSelectQueryAstTranslator`) and mutattion query (`MongoMutationQuerySqlAstTranslator`) respectively.

In this way we can focus on our customization with minimal distraction and make it easy to upgrade Hibernate (as demonstrated by recent v6.6.0 upgrade).

The naming is not perfect and it is convaluted. Initially I made changes directly on the single copied class but soon found it is not maintainable so I move all Mongo customization logic out of it and let it evolve easily.